packs/nagios/etc/st2service_handler.py

```diff
@@ -43,152 +47,291 @@
+        sys.exit(1)
+
+    with open(config_file) as f:
+        config = yaml.safe_load(f)
+
+        if verbose:
+            print('Contents of config file: {0}'.format(config))
+
+        ST2_USERNAME = config['st2_username']
+        ST2_PASSWORD = config['st2_password']
+        ST2_API_KEY = config['st2_api_key']
+        ST2_API_BASE_URL = config['st2_api_base_url']
+        if not ST2_API_BASE_URL.endswith('/'):
+            ST2_API_BASE_URL += '/'
+        ST2_AUTH_BASE_URL = config['st2_auth_base_url']
+        if not ST2_AUTH_BASE_URL.endswith('/'):
+            ST2_AUTH_BASE_URL += '/'
+
+    if ST2_API_KEY:
+        IS_API_KEY_AUTH = True
+
+    if verbose:
+        print('Unauthed? : {0}\nAPI key auth?: {1}\nSSL Verify? : {2}\n'
+              .format(UNAUTHED, IS_API_KEY_AUTH, ST2_SSL_VERIFY))
+
+    if not UNAUTHED and not IS_API_KEY_AUTH:
+        try:
+            if not ST2_AUTH_TOKEN:
+                if verbose:
+                    print('No auth token found. Let\'s get one from'
+                          'StackStorm!')
+                ST2_AUTH_TOKEN = _get_auth_token(verbose=verbose)
+        except:
+            traceback.print_exc(limit=20)
+            print('Unable to negotiate an auth token. Exiting!')
             sys.exit(1)
 
-        with open(ST2_CONFIG_FILE) as f:
-            config = yaml.safe_load(f)
-            ST2_USERNAME = config['st2_username']
-            ST2_PASSWORD = config['st2_password']
-            ST2_API_BASE_URL = config['st2_api_base_url']
-            ST2_AUTH_BASE_URL = config['st2_auth_base_url']
 
-        if not REGISTERED_WITH_ST2:
-            _register_with_st2()
-    except:
-        sys.stderr.write('Failed registering with st2. Won\'t post event.\n')
-    else:
-        main(sys.argv)
+def main(config_file, payload, verbose=False, unauthed=False, ssl_verify=False):
+
+    _set_config_opts(config_file=config_file, unauthed=unauthed,
+                     verbose=verbose, ssl_verify=ssl_verify)
+
+    _register_with_st2(verbose=verbose)
+
+    _post_event_to_st2(payload, verbose=verbose)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='StackStorm nagios event '
+                                     'handler.')
+    parser.add_argument('config_path',
+                        help='Exchange to listen on')
+    parser.add_argument('--verbose', '-v', required=False, action='store_true',
+                        help='Verbose mode.')
+    parser.add_argument('--unauthed', '-u', required=False,
+                        action='store_true', help='Allow to post to '
+                        'unauthed st2. E.g. when auth is disabled ' +
+                        'server side.')
+    parser.add_argument('--ssl-verify', '-s', required=False,
+                        action='store_true', help='Turn on SSL verification '
+                        'for st2 APIs.')
+
+    args = parser.parse_args()
+    payload = sys.stdin.read().strip()
```

Comment:
Does this thing even work? Nagios calls the handlers as ./handler.py <list of args>